http/sys_config_state_test.go

```diff
@@ -9,70 +9,193 @@ import (
 	"testing"
 
 	"github.com/go-test/deep"
+	"github.com/hashicorp/vault/command/server"
+	"github.com/hashicorp/vault/internalshared/configutil"
 	"github.com/hashicorp/vault/vault"
 )
 
 func TestSysConfigState_Sanitized(t *testing.T) {
-	var resp *http.Response
-
-	core, _, token := vault.TestCoreUnsealed(t)
-	ln, addr := TestServer(t, core)
-	defer ln.Close()
-	TestServerAuth(t, addr, token)
-
-	resp = testHttpGet(t, token, addr+"/v1/sys/config/state/sanitized")
-	testResponseStatus(t, resp, 200)
-
-	var actual map[string]interface{}
-	var expected map[string]interface{}
-
-	configResp := map[string]interface{}{
-		"api_addr":                            "",
-		"cache_size":                          json.Number("0"),
-		"cluster_addr":                        "",
-		"cluster_cipher_suites":               "",
-		"cluster_name":                        "",
-		"default_lease_ttl":                   json.Number("0"),
-		"default_max_request_duration":        json.Number("0"),
-		"disable_cache":                       false,
-		"disable_clustering":                  false,
-		"disable_indexing":                    false,
-		"disable_mlock":                       false,
-		"disable_performance_standby":         false,
-		"disable_printable_check":             false,
-		"disable_sealwrap":                    false,
-		"experiments":                         nil,
-		"raw_storage_endpoint":                false,
-		"detect_deadlocks":                    "",
-		"introspection_endpoint":              false,
-		"disable_sentinel_trace":              false,
-		"enable_ui":                           false,
-		"log_format":                          "",
-		"log_level":                           "",
-		"max_lease_ttl":                       json.Number("0"),
-		"pid_file":                            "",
-		"plugin_directory":                    "",
-		"plugin_file_uid":                     json.Number("0"),
-		"plugin_file_permissions":             json.Number("0"),
-		"enable_response_header_hostname":     false,
-		"enable_response_header_raft_node_id": false,
-		"log_requests_level":                  "",
+	cases := []struct {
+		name                    string
+		storageConfig           *server.Storage
+		haStorageConfig         *server.Storage
+		expectedStorageOutput   map[string]interface{}
+		expectedHAStorageOutput map[string]interface{}
+	}{
+		{
+			"raft storage",
```

Comment:
nit: these test cases would be a bit more readable if we used named struct fields, e.g. 
```go
{
  name: "raft storage",
  storageConfig: &server.Storage{...},
  ...
}
```

Reply:
Good point, I'll change that. Thanks for the feedback!